addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/CacheController.java

```diff
@@ -242,11 +250,32 @@ private void cacheBusterHttpMethod(String url, Method method, Cache cache)
                 if (msg.getResponseHeader().getStatusCode()
                         == base.getResponseHeader().getStatusCode()) {
                     String indicValue = msg.getResponseHeader().getHeader(cache.getIndicator());
+                    if (indicValue == null) {
+                        switch (method) {
+                            case GET:
+                                headers.setMethod(HttpRequestHeader.GET);
+                                break;
+                            case POST:
+                                headers.setMethod(HttpRequestHeader.POST);
+                                break;
+                            default:
+                                throw new IllegalArgumentException(
+                                        "Method " + method + " not supported.");
+                        }
```

Comment:
Does not need to be in this PR, letting the enum update the request header would reduce duplication and simplify the code.